### ISSUE TITLE
Wrong sub-/superscript alignment with accents

### ISSUE THREAD

**Issue by EpicEricEE** [typst/typst#1672]
### Description

When using accents like `hat`, the positions of sub- and superscripts change, so that the spacing is off:

```typst
$f_a^a != hat(f)_a^a != hat(f_a^a)$
```

![image](https://github.com/typst/typst/assets/7191192/e5b995d0-b4ec-4698-869c-31a73288dad4)

### Operating system

Web app

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
Only the middle one is what  I would consider a bug

**Comment by EpicEricEE** [comment]
Sure, the right one looks mostly as I would expect. I was only surprised that the superscript was moved down slightly, but I guess that's on purpose to make room for the hat.

**Comment by astrale-sharp** [comment]
![image](https://github.com/typst/typst/assets/53686698/246315cb-43cf-486d-bddb-cd635f76ee58)

this might help `#let y = 0.2em
$f_a^a != hat(f)_a^a != hat(f_a^a)$
#place[#line(stroke: 0.2pt, start: (0pt,y), end: (90pt,y))]
`

**Comment by Enivex** [comment]
Here's an example of LaTeX doing this

![image](https://github.com/typst/typst/assets/2191754/c22759d3-c825-4c83-a7af-e5ed053e2db9)

The placement of scripts is a tricky balance.

**Comment by ludwig-austermann** [comment]
> Here's an example of LaTeX doing this
> 
> ![image](https://github.com/typst/typst/assets/2191754/c22759d3-c825-4c83-a7af-e5ed053e2db9)
> 
> The placement of scripts is a tricky balance between many things.

I don't think that the LaTeX one here is intended nor perfect behaviour, it just is not implemented for the LaTeX package `amsfonts`. If you compare
```LaTeX
$e_1^1 \, \hat{\hat{e}}_1^1$
```
I wouldn't say it is intended behaviour either, but obviously one can not handle every case the desired way, as users may write arbitrary complex expressions.

I think that commonly used symbols which take accents should have commen sub-/ superscripts, as demonstrated here, but also for e.g. double stroke letters. It would be nice to have these common symbols look similar in this respect, when writing (mathematical) function definitions or in inline formulae.

**Comment by Leedehai** [comment]
Nice! Does this fix #3043 ? If so, could you please add a test case for it as well?
